data_steward/validation/main.py

```diff
@@ -882,53 +883,68 @@ def _is_string_excluded_file(gcs_file_name):
-            item_root = item['name'].split('/')[0] + '/'
-            if item_root.lower() in common.IGNORE_DIRECTORIES:
-                ignored_items += 1
-            else:
-                filtered_bucket_items.append(item)
-
-        logging.info(f"Ignoring {ignored_items} items in {hpo_bucket}")
-
-        prefix = f'{hpo_id}/{hpo_bucket}/'
+    except NotFound:
+        logging.warning(f"Unable to acquire HPO '{hpo_id}' bucket using GCS")
+        return
+    # Attempt to acquire drc bucket, raises google.cloud.exceptions.NotFound
+    try:
+        drc_bucket = storage_client.get_drc_bucket()
+    except NotFound:
+        logging.warning('Unable to acquire DRC bucket using GCS')
+        return
 
-        for item in filtered_bucket_items:
-            item_name = item['name']
-            source_blob = source_bucket.get_blob(item_name)
-            destination_blob_name = f'{prefix}{item_name}'
-            source_bucket.copy_blob(source_blob, destination_bucket,
-                                    destination_blob_name)
+    # Attempt to get item metadata
+    # Raises exception BucketDoesNotExistError, HttpError
+    try:
+        bucket_items: list = list_bucket(hpo_bucket.name)
     except BucketDoesNotExistError as bucket_error:
-        bucket = bucket_error.bucket
+        bucket_name: str = bucket_error.bucket
         # App engine converts an env var set but left empty to be the string 'None'
-        if bucket and bucket.lower() != 'none':
+        if bucket_name and bucket_name.lower() != 'none':
             logging.warning(
-                f"Bucket '{bucket}' configured for hpo_id '{hpo_id}' does not exist"
+                f"Bucket '{bucket_name}' configured for hpo_id '{hpo_id}' does not exist"
             )
         else:
             logging.info(
-                f"Bucket '{bucket}' configured for hpo_id '{hpo_id}' is empty/unset"
+                f"Bucket '{bucket_name}' configured for hpo_id '{hpo_id}' is empty/unset"
             )
+        return
     except HttpError as http_error:
-        message = (
+        message: str = (
             f"Failed to copy files for hpo_id '{hpo_id}' due to the following "
             f"HTTP error: {http_error.content.decode()}")
         logging.exception(message)
+        return
+
+    # Filter
+    ignored_count: int = 0
+    allowed_items: list = []
+    for item in bucket_items:
+        item_root: str = item['name'].split('/')[0] + '/'
+        if item_root.lower() in common.IGNORE_DIRECTORIES:
+            ignored_count += 1
+        else:
+            allowed_items.append(item)
+
+    logging.info(f"Ignoring {ignored_count} items in {hpo_bucket.name}")
+    prefix: str = f'{hpo_id}/{hpo_bucket.name}/'
+
+    # Copy
+    for item in allowed_items:
+        name = item['name']
+        hpo_blob = hpo_bucket.get_blob(name)
+        if hpo_blob is not None:
+            full_name: str = f'{prefix}{name}'
+            hpo_bucket.copy_blob(hpo_blob, drc_bucket, full_name)
```

Comment:
I would delete the if statement in line 945 since allowed_items is a filtered list coming from bucket_items in line 908.

Reply:
Yeah.  I can see that.  I think I reasoned that if you capture a blob that doesn't exist, you will get something like "NoneType doesn't copy_blob() method."   Bucket's that grab blobs that don't exist, "fail" silently, and return None. But you're correct; you have a guarantee.  I'll shuffle some around and remove the if block.